lucene/queryparser/build.gradle

```diff
@@ -7,3 +7,224 @@ dependencies {
+            "      this.currentToken = currentTokenVal;${lineSeparator}" +
+            "      this.expectedTokenSequences = expectedTokenSequencesVal;${lineSeparator}" +
+            "      this.tokenImage = tokenImageVal;${lineSeparator}" +
+            "  }",
+        flags: "gm",
+        byline: "false")
+    ant.replaceregexp(file:"src/java/org/apache/lucene/queryparser/flexible/standard/parser/ParseException.java",
+    match: "^  public ParseException\\(String message.*\$(\\s\\s[^}].*\\n)*  \\}",
+    replace: "  public ParseException(Message message) {${lineSeparator}" +
+    "    super(message);${lineSeparator}" +
+    "  }",
+    flags: "gm",
+    byline: "false")
+    ant.replaceregexp(file:"src/java/org/apache/lucene/queryparser/flexible/standard/parser/ParseException.java",
+    match: "^  public ParseException\\(\\).*\$(\\s\\s[^}].*\\n)*  \\}",
+    replace: "  public ParseException() {${lineSeparator}" +
+    "    super(new MessageImpl(QueryParserMessages.INVALID_SYNTAX, \"Error\"));${lineSeparator}" +
+    "  }",
+    flags: "gm",
+    byline: "false")
+    ant.replaceregexp(file: "src/java/org/apache/lucene/queryparser/flexible/standard/parser/ParseException.java",
+    match: "^  public String getMessage\\(\\).*\$(\\s\\s\\s\\s[^}].*\n)*    \\}",
+    replace: "  private static String initialise(Token currentToken, int[][] expectedTokenSequences, String[] tokenImage) {${lineSeparator}" +
+    "String eol = System.getProperty(&quot;lineSeparator&quot;, &quot;\n&quot;);",
+    flags: "gm",
+    byline: "false")
+    ant.replaceregexp(file: "src/java/org/apache/lucene/queryparser/flexible/standard/parser/ParseException.java",
+    match: "\\s*protected String add_escapes.*",
+    replace: "  static private String add_escapes(String str) {",
+    flags: "g",
+    byline: "true")
+  }
+
+  task runJavacc {
+    outputs.upToDateWhen { false } //nocommit
+    group = 'Build Regenerate'
+    description = "Regenerates javacc generated src files."
+    dependsOn runJavaccQueryParser, runJavaccSurround, runJavaccFlexible
+  }
+
+  task regenerate {
+    group = 'Build Regenerate'
+    description = "Regenerates various generated src files, automoton, packedints, jflex, javacc, etc"
+    dependsOn runJavacc
+  }
+}
+
+class JavaCC extends DefaultTask {
+
+  @InputFile
+  File inputFile
+
+  @OutputDirectory
+  File target
+
+  String lineSeparator = System.lineSeparator()
+  @TaskAction
+  void javacc() {
+
+
+    String javaCCClasspath = project.project(":lucene:queryparser").configurations.javaCCDeps.asPath
+    String javaCCHome = javaCCClasspath.substring(0, javaCCClasspath.lastIndexOf("/"))
+
+    // This bit seems really awkward, but I didn't find a good way to either convince the ant task to accept a different
+    // name than javacc.jar...
+    // nocommit So I'm taking the javacc-5.0.jar file that's downloaded to Gradle's cache and renaming it.
```

Comment:
You shouldn't be doing anything with those files. I think it'd be better to use javacc directly rather than through ant - then you could just invoke the javacc with the classpath you declared in the dependency, that's it.

Reply:
OK,I'll try it that way.

Reply:
Guess what? I found a plugin for Gradle that runs and resolves the javacc task, it'll be in the next PR.

Reply:
I know there are plugins for that. We should stick to the exact javacc version ant currently uses though (so that the output code is the same). By the way - the shuffled generated code may be a result of a HashMap somewhere in the javacc code. Sigh.